src/app/clusters/bindings/bindings.cpp

```diff
@@ -0,0 +1,156 @@
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Binding Server Cluster
+ ***************************************************************************/
+
+#include "af.h"
+
+#include <app/util/binding-table.h>
+#include <support/logging/CHIPLogging.h>
+
+using namespace chip;
+
+EmberStatus prepareBinding(EmberBindingTableEntry & binding, NodeId nodeId, GroupId groupId, EndpointId endpointId,
+                           ClusterId clusterId)
+{
+    if (groupId && nodeId)
+    {
+        return EMBER_BAD_ARGUMENT;
+    }
+
+    if (groupId)
+    {
+        binding.type      = EMBER_MULTICAST_BINDING;
+        binding.groupId   = groupId;
+        binding.clusterId = clusterId;
+        binding.local     = emberAfCurrentCommand()->apsFrame->destinationEndpoint;
+    }
+    else
+    {
+        binding.type      = EMBER_UNICAST_BINDING;
+        binding.nodeId    = nodeId;
+        binding.clusterId = clusterId;
+        binding.local     = emberAfCurrentCommand()->apsFrame->destinationEndpoint;
+        binding.remote    = endpointId;
+    }
+
+    return EMBER_SUCCESS;
+}
+
+EmberStatus getBindingIndex(EmberBindingTableEntry & newEntry, uint8_t * bindingIndex)
+{
+    EmberBindingTableEntry currentEntry;
+    for (uint8_t i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
+    {
+        emberGetBinding(i, &currentEntry);
+        if (currentEntry.type == EMBER_UNUSED_BINDING)
+        {
+            continue;
+        }
+
+        if (currentEntry.type == newEntry.type && currentEntry.clusterId == newEntry.clusterId &&
+            currentEntry.local == newEntry.local)
+        {
+            if (currentEntry.type == EMBER_MULTICAST_BINDING && currentEntry.groupId != newEntry.groupId)
+            {
+                continue;
+            }
+
+            if (currentEntry.type == EMBER_UNICAST_BINDING &&
+                (currentEntry.nodeId != newEntry.nodeId || currentEntry.remote != newEntry.remote))
+            {
+                continue;
+            }
+
+            *bindingIndex = i;
+            return EMBER_SUCCESS;
+        }
+    }
+
+    return EMBER_NOT_FOUND;
+}
+
+EmberStatus getUnusedBindingIndex(uint8_t * bindingIndex)
+{
+    EmberBindingTableEntry currentEntry;
+    for (uint8_t i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
+    {
+        emberGetBinding(i, &currentEntry);
+        if (currentEntry.type == EMBER_UNUSED_BINDING)
+        {
+            *bindingIndex = i;
+            return EMBER_SUCCESS;
+        }
+    }
+
+    return EMBER_NOT_FOUND;
+}
+
+bool emberAfBindingClusterBindCallback(NodeId nodeId, GroupId groupId, EndpointId endpointId, ClusterId clusterId)
+{
+    ChipLogDetail(Zcl, "RX: BindCallback");
+
+    EmberBindingTableEntry bindingEntry;
+    if (prepareBinding(bindingEntry, nodeId, groupId, endpointId, clusterId) != EMBER_SUCCESS)
+    {
+        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_MALFORMED_COMMAND);
+        return true;
```

Comment:
since this seems to always return true, do we need it to have a bool return type?

same for below.

Reply:
`bool` is the default value for generated callbacks. For instance you can noticed that the definition of the cluster itself does not specify any return types: https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap-templates/zcl/binding-cluster.xml#L27